md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDAO.java

```diff
@@ -702,5 +725,155 @@ else  if (childProcess.getProcessType().getProcessTypeId() == 36){
+    List<UserRoles> userRoles = criteria.list();
+    List<String> userRolesNameList=new ArrayList<>();
+    String processCreater=process.getUserRoles().getRole();
+    for(UserRoles userRoles1:userRoles)
+    {
+        userRolesNameList.add(userRoles1.getRole());
+    }
 
+    session.getTransaction().commit();
+    session.close();
+    List<Integer> readList=new ArrayList<>();
+    readList.add(4);
+    readList.add(5);
+    readList.add(6);
+    readList.add(7);
+    List<Integer> writeList=new ArrayList<>();
+    writeList.add(2);
+    writeList.add(3);
+    writeList.add(6);
+    writeList.add(7);
+    List<Integer> executeList=new ArrayList<>();
+    executeList.add(1);
+    executeList.add(3);
+    executeList.add(5);
+    executeList.add(7);
+    if (process.getUsers().getUsername().equals(username))
+    {
+        switch (action) {
+            case "write": if (writeList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId())
+                    || (userRolesNameList.contains(processCreater)&&writeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&writeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {
+                return "ACCESS GRANTED";
+            }
+            else
+            {
+                LOGGER.info("user write");
+                throw new SecurityException("ACCESS DENIED");
+            }
+            case "read": if (readList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId()) ||
+                    (userRolesNameList.contains(processCreater)&&readList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&readList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {
+                return "ACCESS GRANTED";
+            }
+            else
+            {
+                LOGGER.info("user read");
+                throw new SecurityException("ACCESS DENIED");
+            }
+            case "execute": if (executeList.contains(process.getPermissionTypeByUserAccessId().getPermissionTypeId())||
+                    (userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByGroupAccessId().getPermissionTypeId()))
+                    || (!userRolesNameList.contains(processCreater)&&executeList.contains(process.getPermissionTypeByOthersAccessId().getPermissionTypeId()))
+                    )
+            {
+                return "ACCESS GRANTED";
+            }
+            else
+            {
+                LOGGER.info("user execute");
+                throw new SecurityException("ACCESS DENIED");
+            }
+            default: {LOGGER.info("no operation");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "no operation" 6 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested code block into a method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1199)